src/components/x-date-range-picker/index.ts

```diff
@@ -0,0 +1,174 @@
+import { render } from 'lit-html';
+import { MobxLitElement } from '@adobe/lit-mobx';
+import { html, customElement, property } from 'lit-element';
+
+import '@vaadin/vaadin-item';
+import '@vaadin/vaadin-select';
+import '@vaadin/vaadin-list-box';
+import '@vaadin/vaadin-date-picker';
+
+import { SelectValueChanged } from '@vaadin/vaadin-select';
+import { DatePickerValueChanged } from '@vaadin/vaadin-date-picker';
+
+import { ValueChangedEvent } from './events';
+
+import {
+  today,
+  weeksAgo,
+  serializeDateRange,
+  deserializeDateRange,
+} from './helpers';
+
+@customElement('x-date-range-picker')
+export class XDateRangePickerElement extends MobxLitElement {
+  /**
+   * The value for this element.
+   *
+   * `"YYYY-MM-DD|YYYY-MM-DD"`
+   *
+   * Supports only the ISO 8601 date format and should use the delimiter
+   * defined in the `delimiter` property (default: `|`).
+   */
+  @property({ type: String })
+  value = '';
+
+  /**
+   * The delimiter that is used to split date range into
+   * `start` date and `end` date.
+   */
+  @property({ type: String })
+  delimiter = '|';
+
+  /**
+   * An array that contains pre-defined date ranges.
+   *
+   * Should match the following structure:
+   * ```
+   * [
+   *  { title: 'Last 2 weeks', value: '14-07-2021|28-07-2021' },
+   *  { title: 'Last 4 weeks', value: '01-07-2021|28-07-2021' },
+   * ]
+   * ```
+   *
+   * The `value` property supports only the ISO 8601 date format and should use the delimiter
+   * defined in the `delimiter` property (default: `|`).
+   */
+  @property({ type: Array })
+  ranges = [
+    {
+      title: 'Last 2 weeks',
+      value: serializeDateRange([weeksAgo(2), today()], this.delimiter),
+    },
+    {
+      title: 'Last 4 weeks',
+      value: serializeDateRange([weeksAgo(4), today()], this.delimiter),
+    },
+  ];
+
+  get startValue() {
+    return deserializeDateRange(this.value, this.delimiter)[0];
+  }
+
+  get endValue() {
+    return deserializeDateRange(this.value, this.delimiter)[1];
+  }
+
+  get selectValue() {
+    const range = this.ranges.find(({ value }) => {
+      return this.value === value;
+    });
+
+    // Check if a pre-defined range is selected
+    if (range) {
+      return range.value;
+    }
+
+    return '';
+  }
+
+  get isCustomDateDisabled() {
+    return this.selectValue !== '';
+  }
+
+  onStartValueChanged(event: DatePickerValueChanged) {
+    const { value } = event.detail;
+
+    this.value = serializeDateRange([value, this.endValue], this.delimiter);
+
+    if (this.startValue && this.endValue) {
+      this.dispatchEvent(new ValueChangedEvent({ value }));
+    }
```

Comment:
Two notes regarding the event:

1. This logic is duplicated here and in other event listeners and always called after a value changes. Consider moving it to the `updated` hook to not repeat yourself:

```ts
updated(props: PropertyValues) {
  if (props.has('value') {
    if (this.startValue && this.endValue) {
      // if you need old value, use `props.get('value')`
      this.dispatchEvent(new ValueChangedEvent({ value: this.value }));
    }
  }
}
```

2. Aside from better readability, this should help to ensure event is only fired once when a pre-defined range is selected and `value` property is assigned to both date-pickers.